pkg/sql/logictest/testdata/logic_test/reassign_owned

```diff
@@ -0,0 +1,22 @@
+statement ok
+CREATE TABLE t()
+
+# Ensure old role must exist.
```

Comment:
Still need to look at the edge cases of what should/should not be possible with this command. Once I do so, can add further cases in either in this PR or the next.

Reply:
Will write more robust logic tests in the follow-up PR after the implementation of the command.